crates/pathfinder/src/rpc/types.rs

```diff
@@ -67,10 +67,12 @@ pub mod request {
     pub struct OverflowingStorageAddress(#[serde_as(as = "H256AsNoLeadingZerosHexStr")] pub H256);
 
     /// Contains parameters passed to `starknet_call`.
+    #[serde_as]
     #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
     #[serde(deny_unknown_fields)]
     pub struct Call {
         pub contract_address: ContractAddress,
+        #[serde_as(as = "Vec<CallParamAsDecimalStr>")]
```

Comment:
Good point. I did this initially because the `starknet` CLI does use decimal strings when calling the `add_transaction` API endpoint.

Will check if it works with hex representation and if it does we can simplify these I guess.

Reply:
No, I don't think it supports hex strings. Trying to use the add_transaction API with hex signatures and calldata leads to this error:
```
StarknetError(StarknetError { code: MalformedRequest, message: "{'signature': {0: ['Expected int string, got: \"0x7dd3a55d94a0de6f3d6c104d7e6c88ec719a82f4e2bbc12587c8c187584d3d5\".'], ...
```

Reply:
Afaik signatures have remained `dec`, while it's only the `CallParam` that does work as `hex`. Maybe you could try with signatures still as `dec`, but only the latter as `hex`?

Reply:
It seems that call parameters as hex work for `add_transaction` -- both for invoke and deploy.

I've removed the corresponding `serde_as()` magic.